Src/FluentAssertions/Primitives/StringWildcardMatchingStrategy.cs

```diff
@@ -2,39 +2,32 @@
 using System.Text;
 using System.Text.RegularExpressions;
 using FluentAssertions.Common;
+using FluentAssertions.Execution;
 
 namespace FluentAssertions.Primitives;
 
-internal class StringWildcardMatchingValidator : StringValidator
+internal class StringWildcardMatchingStrategy : IStringComparisonStrategy
 {
-    public StringWildcardMatchingValidator(string subject, string expected, string because, object[] becauseArgs)
-        : base(subject, expected, because, becauseArgs)
+    public void ValidateAgainstMismatch(IAssertionScope assertion, string subject, string expected)
     {
-    }
-
-    protected override void ValidateAgainstMismatch()
-    {
-        bool isMatch = IsMatch();
+        bool isMatch = IsMatch(subject, expected);
 
-        if (!isMatch && !Negate)
-        {
-            Assertion.FailWith(ExpectationDescription + "but {1} does not.", Expected, Subject);
-        }
-
-        if (isMatch && Negate)
-        {
-            Assertion.FailWith(ExpectationDescription + "but {1} matches.", Expected, Subject);
-        }
+        assertion
+            .ForCondition(!Negate || !isMatch)
+            .FailWith(ExpectationDescription + "but {1} matches.", expected, subject)
+            .Then
+            .ForCondition(Negate || isMatch)
+            .FailWith(ExpectationDescription + "but {1} does not.", expected, subject);
```

Comment:
Let's keep this method as it was.
I haven't dug into the history, but I suspect it was constructed as it was for performance reasons.

Reply:
Changed in [b34b47b](https://github.com/fluentassertions/fluentassertions/pull/2296/commits/b34b47b567e800470868e19b80901b4ee013f0c6)